packages/icon/package.py

```diff
@@ -167,8 +167,7 @@ class Icon(AutotoolsPackage):
     depends_on('libcdi-pio+mpi', when='+cdi-pio+mpi')
 
     # needed to mimick BB in test.py
-    # only available in icon-exclaim so far
-    depends_on('cdo', when='@exclaim-master,exclaim-test')
+    depends_on('cdo')
```

Comment:
This defaults to a build and link-time dependency. But cdo is a runtime dependency. Specifically it is a test-time dependency.
A link-time dependency causes cdo to use the same compiler as icon, but that's an unnecessary restriction for cdo. If it is a run-time or test-time dependency, the concretizer is allowed to mix compilers. I think this is exactly what's needed here.
```suggestion
    depends_on('cdo', type='test')
```